dashboard/app/views/levels/levelGroup.js

```diff
@@ -22,62 +22,35 @@ function initLevelGroup(
     }
   }
 
+  /**
+   * Construct an array of all the level results. When submitted it's something
+   * like this:
+   *
+   * [{"level_id":1977,"result":"0"},{"level_id":2007,"result":"3"},{"level_id":1939,"result":"2,1"}]
+   */
   window.getResult = function () {
-    // Construct an array of all the level results.
-    // When submitted it's something like this:
-    //
-    //   [{"level_id":1977,"result":"0"},{"level_id":2007,"result":"3"},{"level_id":1939,"result":"2,1"}]
-    //
-
     // Add any new results to the existing lastAttempt results.
-    for (var i = 0; i < levelCount; i++) {
-      var levelName = "level_" + i;
-      var levelResult = window[levelName].getCurrentAnswer().toString();
-      var levelId = window[levelName].getLevelId();
-
-      // But before storing, if we had a previous result for the same level,
-      // remove that from the array, since we want to overwrite that previous
-      // answer.
-      for (var j = lastAttempt.length - 1; j >= 0; j--) {
-        if (lastAttempt[j].level_id == levelId) {
-          lastAttempt.splice(j, 1);
-        }
-      }
-
-      lastAttempt.push({level_id: levelId, result: levelResult});
-    }
-
-    var response = JSON.stringify(lastAttempt);
-
+    var results = window.levelGroup.results;
+    Object.keys(results).forEach(function (levelId) {
+      var levelResult = results[levelId].getCurrentAnswer().toString();
+      lastAttempt[levelId] = {result: levelResult};
+    });
 
     var forceSubmittable = window.location.search.indexOf("force_submittable") !== -1;
 
-    var result;
-    var submitted;
-
-    if (window.appOptions.level.submittable || this.forceSubmittable) {
-      result = true;
-      submitted = true;
-    } else {
-      result = true; // this.validateAnswers();
-      submitted = false;
-    }
-
     return {
-      "response": response,
+      "response": JSON.stringify(lastAttempt),
       "result": true,
       "errorType": null,
-      "submitted": submitted
+      "submitted": window.appOptions.level.submittable || forceSubmittable
```

Comment:
Ah I see, should I remove `this.forceSubmittable` entirely?  I was wondering how the UI test passed before this change.